pkg/types/types.go

```diff
@@ -188,3 +191,63 @@ type Service interface {
 	Publisher
 	Querier
 }
+
+// RetrievalRequest is all the details needed for retrieving data from the
+// network. At minimum it contains the URL to retrieve a blob from.
+//
+// Legacy retrievals will not carry any retrieval authorization information i.e.
+// the Auth field will be nil.
+type RetrievalRequest struct {
+	// URL where the blob may be requested from.
+	URL *url.URL
+	// Optional byte range to request.
+	Range *metadata.Range
+	// Optional UCAN authorization parameters.
+	Auth *RetrievalAuth
+}
+
+// NewRetrievalRequest creates a new [RetrievalRequest] object that contains all the
+// details required to retrieve a blob from the network.
+func NewRetrievalRequest(
+	url *url.URL,
+	byteRange *metadata.Range,
+	auth *RetrievalAuth,
+) RetrievalRequest {
+	return RetrievalRequest{url, byteRange, auth}
+}
+
+// RetrievalAuth are the details for a UCAN authorized content retrieval.
+//
+// The provided proofs are expected to contain the `space/content/retrieve`
+// delegated capability allowing content to be retrieved using UCAN
+// authorization.
+type RetrievalAuth struct {
+	// The Indexing Service UCAN signing key.
+	Issuer ucan.Signer
+	// Identity of the storage node to retrieve data from.
+	Audience ucan.Principal
+	// Retrieval ability, resource (typially the space) and caveats.
+	Capability ucan.Capability[ucan.CaveatBuilder]
+	// Delegations from the client to the indexing service authorizing retrieval.
```

Comment:
```suggestion
	// Delegations from the node to the indexing service authorizing retrieval.
```

Reply:
Ah gotcha. I think "the node" is correct - perhaps "the storage node"